core/internal/gethwrappers/abigen.go

```diff
@@ -0,0 +1,57 @@
+package gethwrappers
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	gethParams "github.com/ethereum/go-ethereum/params"
+)
+
+// AbigenArgs is the arguments to the abigen executable. E.g., Bin is the -bin
+// arg.
+type AbigenArgs struct {
+	Bin, ABI, Out, Type, Pkg string
+}
+
+// Abigen calls Abigen  with the given arguments
+//
+// It might seem like a shame, to shell out to another golang program like
+// this, but the abigen executable is the stable public interface to the
+// geth contract-wrapper machinery.
```

Comment:
Wild

Reply:
To be clear, there are "public" interfaces to the underlying golang functions, in the sense that  they're not in `internal` packages and start with capital letters. I just expect that `abigen`'s CLI interface will be more stable than those.